### PR TITLE
[ADD] sale_order_type_confirm_message

### PR DESCRIPTION
**Pull request by Tisho99**
This module includes a validation step when confirming a sale order.

T-7009

### CODE REVIEW

**Comment by Tisho99** [comment]
@HaraldPanten @ValentinVinagre @luis-ron

File: sale_order_type_confirm_message/tests/test_sale_order_type_confirm_message.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class SomethingCase(TransactionCase):
+    def setUp(cls):
```
Comment (rousseldenis) [inline sale_order_type_confirm_message/tests/test_sale_order_type_confirm_message.py]: Please use classmethod instead.

File: sale_order_type_confirm_message/wizards/sale_order_type_confirm_message_wizard.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class SaleOrderTypeConfirmWizard(models.TransientModel):
+    _name = "sale.order.type.confirm.wizard"
+    _description = "Sale Order Type Confirm Message Wizard"
+
+    order_id = fields.Many2one(
+        string="Order",
+        comodel_name="sale.order",
+        default=lambda a: a.env.context.get("active_id", False),
```
Comment (rousseldenis) [inline sale_order_type_confirm_message/wizards/sale_order_type_confirm_message_wizard.py]: `a` -> `wizard` is more significant

File: sale_order_type_confirm_message/tests/test_sale_order_type_confirm_message.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class SomethingCase(TransactionCase):
```
Comment (rousseldenis) [inline sale_order_type_confirm_message/tests/test_sale_order_type_confirm_message.py]: Maybe adding a test with wizard that confirms the sale order (improving coverage)

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by Tisho99** [comment]
@rousseldenis thaks for the review! Requested changes applied

**Review by luis-ron — APPROVED** [review]
Functional review: LGTM 👍🏻

File: sale_order_type_confirm_message/tests/test_sale_order_type_confirm_message.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    def test_sale_order_confirm(self):
+        wizard_form = Form(
+            self.env["sale.order.type.confirm.wizard"].with_context(
+                **{"active_id": self.sale.id}
+            )
+        )
+        wizard_rec = wizard_form.save()
+        res = wizard_rec.action_confirm()
+        self.assertTrue(res)
```
Comment (manuelregidor) [inline sale_order_type_confirm_message/tests/test_sale_order_type_confirm_message.py]: Maybe checking that the sale order is confirmed?

File: sale_order_type_confirm_message/wizards/sale_order_type_confirm_message_wizard_view.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                        <button
+                        string="Cancel"
+                        class="btn-secondary"
+                        special="cancel"
+                    />
+                    </footer>
+            </form>
+        </field>
+    </record>
+    <act_window
```
Comment (manuelregidor) [inline sale_order_type_confirm_message/wizards/sale_order_type_confirm_message_wizard_view.xml]: Is this necessary?

**Review by manuelregidor — CHANGES_REQUESTED** [review]

File: sale_order_type_confirm_message/wizards/sale_order_type_confirm_message_wizard_view.xml (hunk 5 above)
Comment (Tisho99) [reply sale_order_type_confirm_message/wizards/sale_order_type_confirm_message_wizard_view.xml]: No, it wasnt

File: sale_order_type_confirm_message/tests/test_sale_order_type_confirm_message.py (hunk 4 above)
Comment (Tisho99) [reply sale_order_type_confirm_message/tests/test_sale_order_type_confirm_message.py]: Ok, changed

**Review by manuelregidor — APPROVED** [review]
LGTM

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

**Comment by HaraldPanten** [comment]
@rousseldenis Have your requested changes been considered correctly, in order to merge this PR?

THX.

**Review by HaraldPanten — APPROVED** [review]

File: sale_order_type_confirm_message/readme/USAGE.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+To use this module, you need to:
+
+- Create a sale order with a type that has the confirmation message activated and confirm it. Review the warning popup and click accept
```
Comment (HaraldPanten) [inline sale_order_type_confirm_message/readme/USAGE.md]:
```suggestion
- Create a sales order type with the checkbox "Has Confirmation Message" activated, and assign a message in the "Confirmation Message" field.
- Once the user tries to validate a sales quotation that has a confirmation message in it's sales order type, a warning popup will be shown.
- If the user selects the Confirm button, this sales quotation will be confirmed as usual. If the user selects the Cancel button, the popup will be closed and he will go back to the quotation.
```

File: sale_order_type_confirm_message/readme/CONFIGURE.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+To configure this module, you need to:
+
+1. Go to Sales > Configuration > Sales Order Types
+2. Select a Sale Order Type and toggle the "Has Confirmation Message" button
+3. A Confirmation Message field will apear, fill it to include the text that has to be shown on the confirmation screen.
```
Comment (HaraldPanten) [inline sale_order_type_confirm_message/readme/CONFIGURE.md]:
```suggestion
1. Go to Sales > Configuration > Sales Order Types
2. Select a Sale Order Type and toggle the "Has Confirmation Message" button.
3. A Confirmation Message field will apear, fill it to include the text that has to be shown on the confirmation screen.
```

**Review by HaraldPanten — CHANGES_REQUESTED** [review]
Some suggestions to improve it's README

**Review by rousseldenis — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

**Pull request merged**